lib/Doctrine/Common/Proxy/ProxyGenerator.php

```diff
@@ -306,6 +308,17 @@ public function generateProxyClass(ClassMetadata $class, $fileName = false)
         rename($tmpFileName, $fileName);
     }
 
+    private function verifyClassValidnessForProxy(ClassMetadata $class)
```

Comment:
Please add relevant `@throws` annotations in the private/public API of the class being affected by the change